keras_applications/imagenet_utils.py

```diff
@@ -138,12 +141,8 @@ def _preprocess_symbolic_input(x, data_format, mode, **kwargs):
     mean_tensor = backend.constant(-np.array(mean))
 
     # Zero-center by mean pixel
-    if backend.dtype(x) != backend.dtype(mean_tensor):
-        x = backend.bias_add(
-            x, backend.cast(mean_tensor, backend.dtype(x)),
```

Comment:
For some edge cases where input tensor is explicitly set as `uint8`, we are casting the `mean_tensor` to be `uint8` where it should be kept as it is. This is definitely not the intended behaviour.